keps/sig-node/4939-tls-in-grpc-probe/README.md

```diff
@@ -0,0 +1,883 @@
+when reaching out to the health server.
+
+Instead we must continue to rely on the `exec` probe and cannot reap the benefits
+described in [the announcement](https://kubernetes.io/blog/2022/05/13/grpc-probes-now-in-beta/).
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+The primary goal is to support TLS connections when using the `grpc` probe.
+The probe will use TLS but not verify the certificate.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+It is not a goal of this KEP to support providing a certificate to verify the TLS
+connection.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+I would like to add new configuration fields alongside `port` and `service` in the 
+[Probe GRPCAction](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Probe).
+
+They can be used to indicate whether or not TLS should be used.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. Adds more code to Kubelet and surface area to `Pod.Spec`
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We should add a boolean field to the
+[GRPCAction](https://github.com/kubernetes/kubernetes/blob/f422a58398d457a88ac7b05afca85a0e6b805605/pkg/apis/core/types.go#L2665-L2676)
+named `useTLS` which is optional and would default to `false`.
```

Comment:
Proposal looks generally OK.

There are couple of things. Note: I don't think these will change the general design and KEP:

- In HTTP, we chose based on `scheme`, but it looks like gRPC does not have such a notion exposed, so there is no need to push for consistency in naming there.
- gRPC over QUIC only works with TLS. This means if/when we support gRPC over HTTP/3, this will require adding some validation that UseTLS is true.